### ISSUE TITLE
Production grade docker-image

### ISSUE THREAD

**Issue by AtjonTV** [penpot/penpot#301]
I found UXBOX a week ago and I wanted to try it out. But the demo.uxbox.io has some bugs that seem to be fixed in the repo, but setting up a local instance failed multiple times.

The image from the master branch has the issue that it wont build the backend when running the tmux script and the develop branch doesnt find the frontend.

My wish is to be able to build a docker image with the compilation stuff at build time and then have all the services inside to start when the container starts. This would allow for a proper staging/production deployment of UXBOX.

Having to manually start the services with tmux is not a proper way for that kind of deployment.

For what I can see in the demo that I have tested multiple times, UXBOX is a great start at a really usable UX Designer. When the issue I am having with the Docker-Deployment is sorted out I will consider donating to help development.

**Comment by madmath03** [comment]
Hi @AtjonTV ,

Just a side note, but the master branch is quite old and I advise you to check out the develop branch instead.

The devenv docker image in this project is mostly to provide a development environment.
Though it can be used to build prod packages (checkout `manage.sh` [build functions](https://github.com/uxbox/uxbox/blob/5ff0a723d593b5ba8f4961e9fd438d8fd31870ec/manage.sh#L48)), we provide "_production grade_" docker images for UXBOX and we are preparing for the new architecture of the develop branch too :
-  <https://github.com/Monogramm/docker-uxbox-backend>
-  <https://github.com/Monogramm/docker-uxbox-frontend>

Feel free to test those, but do not forget that UXBOX is still in its early stages.

**Comment by AtjonTV** [comment]
I have looked at those functions and run them. But I cant seem to find a way to run it then?

The functions seem to build it locally and not inside the container?

**Comment by niwinz** [comment]
The `manage.sh` functions are responsible to:
- create a development environment (using docker and docker compose)
- use the development image to build prod like bundles ready to deploy to a server.

Is not responsible for:
- generating ready to use docker images
- upload data any server
- neither provide a provisioning

Is just a development helper. If you want to run the development locally, just clone the repo (having docker & docker-compose installed and having the permissions correctly assigned for your user for being able to run docker without `sudo`) and execute:

```bash
./manage.sh build-devenv
./manage.sh run-devenv
```

The first command will start all needed containers, and the second one attaches to one and starts a tmux session with almost all services needed started. You need to know tmux shortcuts for navigate and manage the virtual terminals.

in any case, WARNING: right now the software is in very active development and breaking changes happens, the migrations are not always respected, so you at your own using this for any purpose; this is a development environment

**Comment by AtjonTV** [comment]
Yes I get that the script only builds the prod like stuff locally. But how do I run the builds? I have all the files now but I dont know how to start the thing up ..

This is why I asked for a container that contains the production bundles with a ready-at-start functionality ..

**Comment by niwinz** [comment]
I as responded in other issue, the provisioning and running production builds will be available soon (because we now focused on other parts of the app). But you can use the development environment to know how to run each part:

- For frontend: you need just an http server (nginx, the config you can found on `docker/devenv/files/nginx.conf`) for backend you have `backend/scripts/start-prod` (is outdated but should work).... but apart from all this you will need postgresql and redis and all configuted... and backend should know to which postgresql to connect, etc.. etc.. 

TLDR; If you want hack and look the app, my recommendation is just use the development enviroment (that already starts all the needed parts: nginx, redis, postgresq, backend...) (the default uri is http://localhost:3449/) when all devenv is started).
If you want run yourself production builds, wait until we publish all ansible scripts and/or docker images together with docker-compose...

**Comment by AtjonTV** [comment]
Okay thank you .. I just tried the devenv again and it crashes because it cant find fixtures:

```
uxbox@uxbox-devenv-main:~/uxbox/backend/ > ./scripts/start-dev                  
Loading fixtures...
Execution error (FileNotFoundException) at clojure.main/main (main.java:40).
Could not locate uxbox/fixtures__init.class, uxbox/fixtures.clj or uxbox/fixtures.cljc on classpath.

Full report at:
/tmp/clojure-14982410563952528994.edn
```

But I guess this is just because I am on the develop branch where you just push without CI checks?

I will most likely wait for a release to be tagged and the deploy scripts be published before I will start hosting UXBOX with my other open-use services.
